feat: reset MintLimit to 500K IST in PSMs for USDC, USDT

dckc opened the pull request:
closes: https://github.com/Agoric/dapp-econ-gov/issues/149
refs: https://github.com/Agoric/agoric-private/issues/313

## Description

work-around for #9982 for PSMs, since we don't seem to have access to governor `adminFacet`s.

 - feat(psm): support MintLimit paramUpdates in privateArgs
 - feat: reset MintLimit to 50K IST in all PSMs
    - [x] core eval permit is TODO
 - test: check bundleID between PSM bundle and coreEval

Does **not** support changes to fees.

### Security Considerations

~~Upgrading a PSM, like starting it, requires `feeMintAccess`.~~ As it turns out, `feeMintAccess` is needed to start but not to upgrade.

`instancePrivateArgs` is used as in `packages/vats/src/proposals/upgrade-psm-proposal.js`

### Scaling Considerations

n/a

### Documentation Considerations

 - [x] explanation to BLD stakers
 - [x] explanation to EC

### Testing Considerations

 - [x] [tested on a local a3p chain](#issuecomment-2807825994)
 - [x] bundleID consistency between the contract and the coreEval is checked by a unit test
 - verification using inquisitor and recent mainnet state ran into limitations around vat upgrade in inquisitor

### Upgrade Considerations

upgrades PSMs for USDC, USDT.

mhofman commented on packages/inter-protocol/src/proposals/psm-gov-reset.js:
I'm confused, why can't this file be generated through the normal process of writing a core eval from a proposal using agoric cli ?

Afaik, all files in this folder follow that pattern, no ?

The limits could more easily be made configurable in that case too.

mhofman replied:
I assume not. Main concern is updating the bundle ref. That's one thing that proposal builder takes care of

dckc replied:
yes, that's a limitation of this approach.

mhofman reviewed (commented):
I'm surprised to see a manually authored core-eval instead of a proposal. That would also make it simpler to make the updated params values configurable I think?

dckc commented:
## tested on a local a3p chain

1. build the `29e7a4...` bundle: in `packages/inter-protocol`: `yarn bundle-source --elide-comments --cache-json ./bundles/ src/psm/psm.js psm`
2. start the chain: in a `dapp-agoric-basics` checkout, `docker compose pull` to get current a3p image (hash below); then `yarn start:docker` and watch logs with `yarn docker:logs`
3. using cosgov, install the bundle
4. in `dapp-agoric-basics`, get ready to vote with: `yarn docker:make vote PROPOSAL=24`
5. using the cosgov coreEval form, add `psm-gov-reset-permit.json` and `psm-gov-reset.js`; sign and submit
6. enter the `... make vote ...` command above
7. see `agd` logs as below showing each PSM updated; for example: `v61: updated { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } for Object [Alleged: USDC brand] {}`

<details><summary>coreEval execution logs from agd</summary>

```
agd-1  | 2025-04-16T00:02:05.838Z block-manager: block 1548 commit
agd-1  | 2025-04-16T00:02:06.848Z block-manager: block 1549 begin
agd-1  | 2025-04-16T00:02:06.858Z SwingSet: vat: v1: --- PGRe 1 script started
agd-1  | 2025-04-16T00:02:06.858Z SwingSet: vat: v1: --- PGRe 2 "export" using completion value [Function: resetPSMMintLimits]
agd-1  | 2025-04-16T00:02:06.858Z SwingSet: vat: v1: --- PGRe 3 resetPSMMintLimits { bundleID: 'b1-29e7a47537e6fedccb75dd2c01e28a4bfdf787c6c3c7d7e06156627ed5b31ba80aa05727d10350fa776af0d13eacf15af586da7a8468e9fa0a3f9d41727fc67a' }
agd-1  | 2025-04-16T00:02:06.859Z SwingSet: vat: v1: --- PGRe 4 { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } }
agd-1  | 2025-04-16T00:02:06.860Z SwingSet: vat: v1: --- PGRe 5 PSM name(s) psm-IST-DAI_axl DAI_axl
agd-1  | 2025-04-16T00:02:07.112Z SwingSet: vat: v1: --- PGRe 6 DAI_axl privateArgs { feeMintAccess: Object [Alleged: FeeMint feeMintAccess] {}, initialPoserInvitation: Object [Alleged: Zoe Invitation payment] {}, marshaller: Object [Alleged: Board publishingMarshaller] {}, storageNode: Object [Alleged: ChainStorageNode] {}, paramUpdates: { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } }
agd-1  | 2025-04-16T00:02:07.130Z SwingSet: kernel: attempting to upgrade vat v37 from incarnation 1 to source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:08.745Z SwingSet: vat: v37: PSM Starting Object [Alleged: DAI_axl brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: DAI_axl brand] {}, value: 1_000_000_000_000_000_000n } }
agd-1  | 2025-04-16T00:02:08.848Z SwingSet: kernel: vat v37 upgraded from incarnation 1 to 2 with source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:09.322Z SwingSet: vat: v37: updated { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } for Object [Alleged: DAI_axl brand] {}
agd-1  | 2025-04-16T00:02:09.331Z SwingSet: vat: v1: --- PGRe 7 upgraded DAI_axl { incarnationNumber: 2 }
agd-1  | 2025-04-16T00:02:09.332Z SwingSet: vat: v1: --- PGRe 8 PSM name(s) psm-IST-DAI_grv DAI_grv
agd-1  | 2025-04-16T00:02:09.404Z SwingSet: vat: v1: --- PGRe 9 DAI_grv privateArgs { feeMintAccess: Object [Alleged: FeeMint feeMintAccess] {}, initialPoserInvitation: Object [Alleged: Zoe Invitation payment] {}, marshaller: Object [Alleged: Board publishingMarshaller] {}, storageNode: Object [Alleged: ChainStorageNode] {}, paramUpdates: { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } }
agd-1  | 2025-04-16T00:02:09.418Z SwingSet: kernel: attempting to upgrade vat v38 from incarnation 1 to source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:11.016Z SwingSet: vat: v38: PSM Starting Object [Alleged: DAI_grv brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: DAI_grv brand] {}, value: 1_000_000_000_000_000_000n } }
agd-1  | 2025-04-16T00:02:11.119Z SwingSet: kernel: vat v38 upgraded from incarnation 1 to 2 with source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:11.577Z SwingSet: vat: v38: updated { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } for Object [Alleged: DAI_grv brand] {}
agd-1  | 2025-04-16T00:02:11.585Z SwingSet: vat: v1: --- PGRe 10 upgraded DAI_grv { incarnationNumber: 2 }
agd-1  | 2025-04-16T00:02:11.586Z SwingSet: vat: v1: --- PGRe 11 PSM name(s) psm-IST-USDC_axl USDC_axl
agd-1  | 2025-04-16T00:02:11.659Z SwingSet: vat: v1: --- PGRe 12 USDC_axl privateArgs { feeMintAccess: Object [Alleged: FeeMint feeMintAccess] {}, initialPoserInvitation: Object [Alleged: Zoe Invitation payment] {}, marshaller: Object [Alleged: Board publishingMarshaller] {}, storageNode: Object [Alleged: ChainStorageNode] {}, paramUpdates: { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } }
agd-1  | 2025-04-16T00:02:11.674Z SwingSet: kernel: attempting to upgrade vat v39 from incarnation 1 to source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:13.437Z SwingSet: vat: v39: PSM Starting Object [Alleged: USDC_axl brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: USDC_axl brand] {}, value: 1_000_000n } }
agd-1  | 2025-04-16T00:02:13.544Z SwingSet: kernel: vat v39 upgraded from incarnation 1 to 2 with source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:14.027Z SwingSet: vat: v39: updated { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } for Object [Alleged: USDC_axl brand] {}
agd-1  | 2025-04-16T00:02:14.035Z SwingSet: vat: v1: --- PGRe 13 upgraded USDC_axl { incarnationNumber: 2 }
agd-1  | 2025-04-16T00:02:14.036Z SwingSet: vat: v1: --- PGRe 14 PSM name(s) psm-IST-USDC_grv USDC_grv
agd-1  | 2025-04-16T00:02:14.112Z SwingSet: vat: v1: --- PGRe 15 USDC_grv privateArgs { feeMintAccess: Object [Alleged: FeeMint feeMintAccess] {}, initialPoserInvitation: Object [Alleged: Zoe Invitation payment] {}, marshaller: Object [Alleged: Board publishingMarshaller] {}, storageNode: Object [Alleged: ChainStorageNode] {}, paramUpdates: { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } }
agd-1  | 2025-04-16T00:02:14.131Z SwingSet: kernel: attempting to upgrade vat v40 from incarnation 1 to source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:15.938Z SwingSet: vat: v40: PSM Starting Object [Alleged: USDC_grv brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: USDC_grv brand] {}, value: 1_000_000n } }
agd-1  | 2025-04-16T00:02:16.043Z SwingSet: kernel: vat v40 upgraded from incarnation 1 to 2 with source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:16.490Z SwingSet: vat: v40: updated { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } for Object [Alleged: USDC_grv brand] {}
agd-1  | 2025-04-16T00:02:16.499Z SwingSet: vat: v1: --- PGRe 16 upgraded USDC_grv { incarnationNumber: 2 }
agd-1  | 2025-04-16T00:02:16.500Z SwingSet: vat: v1: --- PGRe 17 PSM name(s) psm-IST-USDT_axl USDT_axl
agd-1  | 2025-04-16T00:02:16.568Z SwingSet: vat: v1: --- PGRe 18 USDT_axl privateArgs { feeMintAccess: Object [Alleged: FeeMint feeMintAccess] {}, initialPoserInvitation: Object [Alleged: Zoe Invitation payment] {}, marshaller: Object [Alleged: Board publishingMarshaller] {}, storageNode: Object [Alleged: ChainStorageNode] {}, paramUpdates: { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } }
agd-1  | 2025-04-16T00:02:16.584Z SwingSet: kernel: attempting to upgrade vat v41 from incarnation 1 to source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:18.234Z SwingSet: vat: v41: PSM Starting Object [Alleged: USDT_axl brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: USDT_axl brand] {}, value: 1_000_000n } }
agd-1  | 2025-04-16T00:02:18.339Z SwingSet: kernel: vat v41 upgraded from incarnation 1 to 2 with source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:18.809Z SwingSet: vat: v41: updated { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } for Object [Alleged: USDT_axl brand] {}
agd-1  | 2025-04-16T00:02:18.816Z SwingSet: vat: v1: --- PGRe 19 upgraded USDT_axl { incarnationNumber: 2 }
agd-1  | 2025-04-16T00:02:18.818Z SwingSet: vat: v1: --- PGRe 20 PSM name(s) psm-IST-USDT_grv USDT_grv
agd-1  | 2025-04-16T00:02:18.884Z SwingSet: vat: v1: --- PGRe 21 USDT_grv privateArgs { feeMintAccess: Object [Alleged: FeeMint feeMintAccess] {}, initialPoserInvitation: Object [Alleged: Zoe Invitation payment] {}, marshaller: Object [Alleged: Board publishingMarshaller] {}, storageNode: Object [Alleged: ChainStorageNode] {}, paramUpdates: { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } }
agd-1  | 2025-04-16T00:02:18.903Z SwingSet: kernel: attempting to upgrade vat v42 from incarnation 1 to source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:20.534Z SwingSet: vat: v42: PSM Starting Object [Alleged: USDT_grv brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: USDT_grv brand] {}, value: 1_000_000n } }
agd-1  | 2025-04-16T00:02:20.643Z SwingSet: kernel: vat v42 upgraded from incarnation 1 to 2 with source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:21.084Z SwingSet: vat: v42: updated { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } for Object [Alleged: USDT_grv brand] {}
agd-1  | 2025-04-16T00:02:21.097Z SwingSet: vat: v1: --- PGRe 22 upgraded USDT_grv { incarnationNumber: 2 }
agd-1  | 2025-04-16T00:02:21.098Z SwingSet: vat: v1: --- PGRe 23 PSM name(s) psm-IST-USDC USDC
agd-1  | 2025-04-16T00:02:21.165Z SwingSet: vat: v1: --- PGRe 24 USDC privateArgs { feeMintAccess: Object [Alleged: FeeMint feeMintAccess] {}, initialPoserInvitation: Object [Alleged: Zoe Invitation payment] {}, marshaller: Object [Alleged: Board publishingMarshaller] {}, storageNode: Object [Alleged: ChainStorageNode] {}, paramUpdates: { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } }
agd-1  | 2025-04-16T00:02:21.180Z SwingSet: kernel: attempting to upgrade vat v61 from incarnation 1 to source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:22.763Z SwingSet: vat: v61: PSM Starting Object [Alleged: USDC brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: USDC brand] {}, value: 1_000_000n } }
agd-1  | 2025-04-16T00:02:22.870Z SwingSet: kernel: vat v61 upgraded from incarnation 1 to 2 with source b1-8bb1b4d998cc4d9eb842ce75b691075c86851b98c48cfa4a955babda63a6a5de6cb44005deb9a065e5d57e4f6b20955fd19f4cf908b79bb5e412807d5431edc1
agd-1  | 2025-04-16T00:02:23.271Z SwingSet: vat: v61: updated { MintLimit: { brand: Object [Alleged: IST brand] {}, value: 500_000_000_000n } } for Object [Alleged: USDC brand] {}
agd-1  | 2025-04-16T00:02:23.279Z SwingSet: vat: v1: --- PGRe 25 upgraded USDC { incarnationNumber: 2 }
agd-1  | 2025-04-16T00:02:23.973Z block-manager: block 1549 commit
agd-1  | 2025-04-16T00:02:24.106Z block-manager: block 1550 begin
agd-1  | 2025-04-16T00:02:24.111Z block-manager: block 1550 commit
agd-1  | 2025-04-16T00:02:25.111Z block-manager: block 1551 begin
```
</details> 


image details:
```
REPOSITORY                          TAG                                          IMAGE ID       CREATED         SIZE
ghcr.io/agoric/agoric-3-proposals   latest                                       a0fe16bb0e27   26 hours ago    4.71GB
```
